### DISCUSSION TITLE
变量初始化

### DISCUSSION THREAD

**Discussion by paprika0741** [arcsysu/SYsU-lang#46] [Q&A]
```
const int N = -1;
 int arr[N + 2 * 4 - 99 / 99] = {1, 2, 33, 4, 5, 6};
int main()
{
return 0;
}
```
+ 数组声明需要知道维数的int， `N + 2 * 4 - 99 / 99` 需要进行运算，得到Value* 需要BuilderCreate ，此时没有函数也没块怎么办呢？
+ Value*如何得到int数值,`llvm::ArrayType::get` 需要int类型
（u1s1从28写到153，中间没有增加使用关于llvm新的东西..../(ㄒoㄒ)/~~/(ㄒoㄒ)/~~/(ㄒoㄒ)/~~/(ㄒoㄒ)/~~/(ㄒoㄒ)/~~/(ㄒoㄒ)/~~）

**Comment by paprika0741** [comment]
能不能求一个评测中间错误不停止的脚本 ，测完所有的统计通过数😢

**Reply by wu-kan** [reply]
本地可以错一个就删掉一个算例？

目前模板仓库里不直接提供这样的脚本，因为希望把错误尽量早暴露出来，如果想要达到这样的功能可能要自己改脚本了？

**Reply by paprika0741** [reply]
> 本地可以错一个就删掉一个算例？
> 
> 目前模板仓库里不直接提供这样的脚本，因为希望把错误尽量早暴露出来，如果想要达到这样的功能可能要自己改脚本了？

上面的问题捏

**Reply by Thirtory** [reply]
> 本地可以错一个就删掉一个算例？
> 
> 目前模板仓库里不直接提供这样的脚本，因为希望把错误尽量早暴露出来，如果想要达到这样的功能可能要自己改脚本了？

有些错误一两个测试点看不出来（点名数组的GEP）多测几个一起看语法树输出的话感觉可以少走点弯路
![012814C0](https://user-images.githubusercontent.com/69828470/169660908-7cc6c349-c05a-47dd-8986-67eaa5e46ba7.png)

**Reply by paprika0741** [reply]
> 

还有好些重复的测例😢

**Reply by paprika0741** [reply]
其实数组的常数语法树里有，全局变量初始化可以转换成int计算..再转换成str...
